File: ScpProject/StrengthConApp/src/app/Models/Athlete.ts
import { ITaggable } from './../Interfaces/ITaggable';
import { Tag } from './Tag';
export class Athlete implements ITaggable {
    public Id: number = 0;
    public FirstName: string;
    public LastName: string;
    public Name: string;
    public Email: string;
    public ProfilePicture: AthletePictures;
    public Weight: number;
    public HeightPrimary: number;
    public HeightSecondary: number;
    public Birthday: Date;
    Tags: Tag[] = [];
    public DisplayTags: DisplayTags[] = [];
    public Checked: boolean = false;//Lazy here. This hack is just to be used on the program check all. When we start cleaning up the code
    //this needs to be changed.
    public UserId: number;
    public ValidatedEmail: boolean;
}

export class AthletePictures {

    public Id: number;
    public URL: string;
    public FileName: string;
    public Thumbnail: string;
    public Profile: string;
}

export class DisplayTags {
    public display: string
    public value: number;
}

File: ScpProject/BL/BusinessObjects/Athlete.cs
using System;
using System.Collections.Generic;
using b = BL.BusinessObjects;
using m = Models;

namespace BL.BusinessObjects
{
    public class Athlete
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int AthleteUserId { get; set; }//the user associated with an athlete
        public int? ProgramId { get; set; }//this overrides a team assigned program
        public double? HeightPrimary { get; set; }
        public double? HeightSecondary { get; set; }
        public DateTime? Birthday { get; set; }
        public double? Weight { get; set; }
        public int UserId { get; set; }

        public string Email { get; set; }
        public List<b.Tag> Tags { get; set; }
        public m.MultiMedia.Picture ProfilePicture { get; set; }
        public bool IsDeleted { get; set; }
        public bool ValidatedEmail { get; set; }

    }
}
